wordpress-mobile/WordPress-Android
# Issue/8670 Site creation error verticals

Fixes #8731 

**Note: `issue/8670-site-creation-errors-segments` is set as the target branch -> the target branch should be changed to `feature/master-site-creation` when #8732 is merged.**

1. Adds support for adding errorMessage**ResourceId** to the ListState, so we can easily use it in ViewModels where we don't have a `Context`.
2. UiState refactoring - I've separated the UiState into three sub states -> Progress/Content/Error -> if a value in the UiState is null -> we hide the view. 
3. Adds support for "No connection" errors

To test:
- set the `NEW_SITE_CREATION_ENABLED` build time feature flag to `true`
1. Start the SiteCreation flow
2. Select a segment
3. Type a query
4. Select a vertical
5. Make sure you are navigated to the domains screen
-------
1. Start the SiteCreation flow
2. Turn on Airplane mode
3. Select a segment
3. Notice the Fullscreen "No connection" error message
4. Click on Retry
5. Notice the Fullscreen "No connection" error message
6. Turn off airplane mode
7. Click on Retry
8. Make sure the verticals screen is shown
9. Turn on airplane mode
10. Type a query 
11. Notice the "No connection" item 
12. Turn off airplane mode
13. Click on retry
14. Notice verticals are shown
--------
- add `event.error = FetchSegmentsError(GENERIC_ERROR)` to `FetchSegmentsUseCase.onSiteCategoriesFetched(...)`
1. Start the SiteCreation flow
2. Select a segment
3. Notice the "There was a problem" error
--------
- add `event.error = FetchVerticalsError(GENERIC_ERROR)` to `FetchVerticalsUseCase.onVerticalsFetched`
1. Start the SiteCreation flow
2. Select a segment
3. Type a query 
4. Notice the "There was a problem" error item
-------
![errors](https://user-images.githubusercontent.com/2261188/49514139-7e70c200-f893-11e8-8c04-7deb16827ee4.png)
